src/toil/statsAndLogging.py

```diff
@@ -231,8 +233,34 @@ def callback(fileHandle: Union[IO[bytes], IO[str]]) -> None:
                 )
                 cls.writeLogFiles(jobNames, messages, config=config)
 
+            try:
+                jobs = stats.jobs
+            except AttributeError:
+                pass
+            else:
+                for job in jobs:
+                    assert config.workflowID is not None
```

Comment:
Should this also be in the `try`/`except`?

Reply:
Maybe? I think here we'll get an actual error and stop the stats and logging thread if the worker sends back job executions and no workflow ID has been set on the config.

But we probably want that thread to stay up even if something is semantically wrong at the worker, because stopping stats and logging won't stop the workflow.